Source/Core/Core/HW/DVD/AMMediaboard.cpp

```diff
@@ -0,0 +1,1829 @@
+        s_media_buffer[6] = 1;          // Type: VxWorks
+        break;
+      case AMMBCommand::GetSystemFlags:
+        s_media_buffer[4] = 1;
+        s_media_buffer[6] = NANDMaskBoardNAND;
+        s_media_buffer[7] = 1;
+        break;
+      // Empty reply
+      case AMMBCommand::Unknown_103:
+        break;
+      case AMMBCommand::Unknown_104:
+        s_media_buffer[4] = 1;
+        break;
+      case AMMBCommand::Accept:
+      {
+        const u32 fd = s_sockets[SocketCheck(s_media_buffer_32[2])];
+        int ret = -1;
+        sockaddr addr;
+        int len = 0;
+
+        // Handle optional parameters
+        if (s_media_buffer_32[3] == 0 || s_media_buffer_32[4] == 0)
+        {
+          ret = NetDIMMAccept(fd, nullptr, nullptr);
+        }
+        else
+        {
+          const u32 addr_off = s_media_buffer_32[3] - NetworkCommandAddress2;
+          const u32 len_off = s_media_buffer_32[4] - NetworkCommandAddress2;
+
+          ret = NetDIMMAccept(fd, &addr, &len);
+          if (len)
+          {
+            memcpy((s_network_command_buffer + addr_off), &addr, len);
+            memcpy((s_network_command_buffer + len_off), &len, sizeof(int));
+          }
+        }
+
+        NOTICE_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: accept( {}({}) ):{}\n", fd, s_media_buffer_32[2],
+                       ret);
+        s_media_buffer_32[1] = ret;
+        break;
+      }
+      case AMMBCommand::Bind:
+      {
+        sockaddr_in addr;
+
+        const u32 fd = s_sockets[SocketCheck(s_media_buffer_32[2])];
+        const u32 off = s_media_buffer_32[3] - NetworkCommandAddress2;
+        const u32 len = s_media_buffer_32[4];
+
+        memcpy((void*)&addr, s_network_command_buffer + off, sizeof(sockaddr_in));
+
+        addr.sin_family = Common::swap16(addr.sin_family);
+
+        // Triforce titles typically rely on hardcoded IP addresses.
+        // This behavior has been modified to bind to the wildcard address instead.
+        //
+        // addr.sin_addr.s_addr = htonl(addr.sin_addr.s_addr);
+
+        addr.sin_addr.s_addr = INADDR_ANY;
+
+        const int ret = bind(fd, (const sockaddr*)&addr, len);
+        const int err = WSAGetLastError();
+
+        if (ret < 0)
+          PanicAlertFmt("Socket Bind Failed with{0}", err);
+
+        NOTICE_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: bind( {}, ({},{:08x}:{}), {} ):{} ({})\n", fd,
+                       addr.sin_family, addr.sin_addr.s_addr, Common::swap16(addr.sin_port), len,
+                       ret, err);
+
+        s_media_buffer_32[1] = ret;
+        s_last_error = SSC_SUCCESS;
+        break;
+      }
+      case AMMBCommand::Closesocket:
+      {
+        const u32 fd = s_sockets[SocketCheck(s_media_buffer_32[2])];
+
+        const int ret = closesocket(fd);
+
+        NOTICE_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: closesocket( {}({}) ):{}\n", fd,
+                       s_media_buffer_32[2], ret);
+
+        s_sockets[s_media_buffer_32[2]] = SOCKET_ERROR;
```

Comment:
Is it possible for `s_media_buffer_32[2]` to be manipulated to produce an out of bound read?

Reply:
We have to protect against the possibility of someone running a homebrew program that has been designed to attack Dolphin. I'm not going to require that the behavior is accurate to the real hardware, but for security reasons we mustn't trigger out-of-bounds memory accesses.